agent/src/main/java/com/microsoft/applicationinsights/agent/internal/config/DataOfConfigurationForException.java

```diff
@@ -28,16 +28,25 @@
  */
 public final class DataOfConfigurationForException {
     private boolean enabled = false;
-    private int stackSize = Integer.MAX_VALUE;
+    private Integer maxStackSize;
+    private Integer maxTraceLength;
     private HashSet<String> suppressedExceptions = new HashSet<String>();
     private HashSet<String> validPathForExceptions = new HashSet<String>();
 
-    public int getStackSize() {
-        return stackSize;
+    public Integer getMaxStackSize() {
```

Comment:
can this method be converted to return primitive `int` instead of Boxed type `Integer`. I do not see any specific need of it.

Reply:
The idea is to avoid the duplication of decision logic for default values (in xml, sb and agent config). It seems more convenient to keep it in one place (see ExceptionTelemetryOptions.java).
Null value, in case of the absence of specific configuration, seems like a natural choice.